lib/beaker/host/unix/exec.rb

```diff
@@ -49,20 +49,20 @@ def reboot(wait_time=10, max_connection_tries=9, uptime_retries=18)
       sleep(reboot_sleep)
 
       exec(Beaker::Command.new('/bin/systemctl reboot -i || reboot || /sbin/shutdown -r now'), :expect_connection_failure => true)
-    rescue Beaker::Host::RebootFailure => e
+    rescue RuntimeError => e
+      raise Beaker::Host::RebootFailure, "Unexpected exception in reboot: #{e.message}"
+    rescue ArgumentError => e
+      raise Beaker::Host::RebootFailure, "Unable to parse time: #{e.message}"
+    rescue => e
+      # Yes, this isn't best practice but rebooting is weird and we really want
+      # to make sure we retry if something went wrong (no matter what it was).
       attempts += 1
       if attempts < uptime_retries
-        @logger.debug("Could not get initial boot time. Will retry #{uptime_retries - attempts} more times.")
+        @logger.warn("Could not get initial boot time. Will retry #{uptime_retries - attempts} more times.")
         retry
       else
         raise
       end
-    rescue Beaker::Host::CommandFailure => e
-      raise Beaker::Host::RebootFailure, "Command failed when attempting to reboot: #{e.message}"
```

Comment:
Do we want to stop explicitly rescuing `Beaker::Host::CommandFailure` exceptions?  The specificity of this message has been helpful when troubleshooting beaker failures.

Reply:
@op-ct It wasn't more helpful than it was confusing in the end honestly. I won't lie, the whole thing probably needs a refactor but, right now, I've got things that are breaking and this implementation has worked across everything I've thrown at it.

Reply:
As far as I can see, the implementation would work just as well if the `Beaker::Host::CommandFailure` was still in place.  I disagree that the message was more confusing than helpful; at least you could narrow a failure down to an attempted host command.  

If there isn't a stronger reason to remove the rescue, I would prefer that it remains.

Reply:
@op-ct I'm 99% sure that the end rescue never gets hit during the command failure. This means that the `retry` never triggers if the SSH tries to hit the system but the system is taking a while to come back up. Honestly, I ended up hitting so many edge cases depending on system load, lag, etc..., that I basically narrowed it down to the 'safe ones' and 'everything else'.